### ISSUE TITLE
Rule proposal: @typescript-eslint/require-atomic-updates

### ISSUE THREAD

**Issue by s-h-a-d-o-w** [typescript-eslint/typescript-eslint#4346]
### Description

 An extension rule for [`require-atomic-updates`](https://eslint.org/docs/rules/require-atomic-updates) that doesn't consider using `typeof` as reading the value.

### Fail

```ts
let foo = 3;

async function incrementFoo() {
  const increment: typeof foo = foo;
  await 0;
  foo += increment; // error
}
```

### Pass

```ts
let foo = 3;

async function incrementFoo() {
  const increment: typeof foo = 3;
  await 0;
  foo += increment;
}
```

**Comment by Josh-Cena** [comment]
As of the latest ESLint version, the base rule no longer reports on this code, so closing as completed

**Comment by s-h-a-d-o-w** [comment]
Awesome!

I would by the way recommend posting a message if you agree with a proposal and want a PR, because I had no idea that I could've worked on this. 😅

**Comment by Josh-Cena** [comment]
Ha! Yeah sometimes we say "accepting a PR doing X", sometimes the issue is so well-articulated that people don't need additional input to understand what to do so we just label them anyway, but labeling doesn't send you a notification. Sometimes I check my created issues just to see their status, because this happens with other repos too.

**Comment by s-h-a-d-o-w** [comment]
That's a great point! I believe GitHub didn't have a convenient way for globally discovering opened issues or PRs for a pretty long time.
But apparently, it finally does. Thanks for making me realize that! 😄

**Comment by Josh-Cena** [comment]
Right, the https://github.com/issues page has been there for a while now 😄

**Comment by s-h-a-d-o-w** [comment]
I better stop talking now... 🙈😆
